Accounting for resources (experimental mode) (#63)
vacp2p/rfc: waku.md

Received envelopes MUST be passed through the Waku envelopes pipelines so that they are picked up by registered filters and passed to subscribers.

### Accounting for resources (experimental)

> This is an experimental mode that MAY be implemented. It is heavily inspired by Swarm's SWAP protocol, and works by doing pairwise accounting for resources.

Reviewer: ```suggestion
Nodes MAY implement accounting, keeping track of resource usage. It is heavily inspired by Swarm's SWAP protocol, and works by doing pairwise accounting for resources.
```

> Each node keeps track of resource usage with all other node. Whenever an envelope is received from a node that is expected (fits bloom filter or topic interest, is legal, etc) this is tracked.

Reviewer: ```suggestion
Each node keeps track of resource usage with all other nodes. Whenever an envelope is received from a node that is expected (fits bloom filter or topic interest, is legal, etc) this is tracked.
```

> Every time epoch (say, every minute or every time an event happens) statistics SHOULD be print of the form:

Reviewer: ```suggestion
Every time epoch (e.g. every minute or every time an event happens) statistics SHOULD be print of the form:
```

> Every time epoch (say, every minute or every time an event happens) statistics SHOULD be print of the form:

Reviewer: printed where?

Author: Um, STDOUT or wherever seems appropriate for implementation specific logging? Don't think this needs to be overly prescriptive

Reviewer: I don't think whether or not nodes print to STDOUT should be part of the spec, nodes should account which is what we care about. We shouldn't care about UI imo.

Author: fair, it isn't meant to be part of the final spec, just experimental placeholder for now while there are no consequences for accounting. We can remove it from the spec and just make sure it happens in clients separately I suppose

Author: > Every time epoch (say, every minute or every time an event happens) statistics SHOULD be print of the form: 

=>

>  Every time epoch (say, every minute or every time an event happens) statistics SHOULD be aggregated and saved by the client

Reviewer: @oskarth yeah that sounds better.

Reviewer 2: yeah, agree here. As there are other means to collect these statistics such as metrics toolkits.

Received envelopes MUST be passed through the Waku envelopes pipelines so that they are picked up by registered filters and passed to subscribers.

### Accounting for resources (experimental)

> Nodes MAY implement accounting, keeping track of resource usage. It is heavily inspired by Swarm's SWAP protocol, and works by doing pairwise accounting for resources.

Reviewer: Link to swap?

Author: ```suggestion
Nodes MAY implement accounting, keeping track of resource usage. It is heavily inspired by Swarm's [SWAP protocol](https://www.bokconsulting.com.au/wp-content/uploads/2016/09/tron-fischer-sw3.pdf), and works by doing pairwise accounting for resources.
```

this?

Author: no canonical link yet IIRC, their specs are in PR

Each node keeps track of resource usage with all other nodes. Whenever an envelope is received from a node that is expected (fits bloom filter or topic interest, is legal, etc) this is tracked.

> Every time epoch (say, every minute or every time an event happens) statistics SHOULD be aggreated and saved by the client:

Reviewer: ```suggestion
Every epoch (say, every minute or every time an event happens) statistics SHOULD be aggregated and saved by the client:
```
